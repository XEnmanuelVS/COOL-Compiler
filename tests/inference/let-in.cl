class Main {
    attr: String <- "Hello world";

    func(): AUTO_TYPE {
        let x: AUTO_TYPE <- attr in
            x.length()
    };
};
