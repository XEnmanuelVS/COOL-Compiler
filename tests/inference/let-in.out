\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class Main  { <feature> ... <feature> }
    \__AttrDeclarationNode: attr : String
    \__FuncDeclarationNode: func() : Int -> <body>
      \__LetNode:  let
       \__VarDeclarationNode: x: String <-
        \__VariableNode: attr
        in
          \__CallNode: <obj>.length(<expr>, ..., <expr>)
            \__VariableNode: x

